ui/src/message_popup/author/questions_page.jsx

```diff
@@ -59,26 +70,54 @@ export default React.createClass({
     return questionString.split(text).length;
   },
   
+  getSearchResults(){
+    const {searchText} = this.state;
+    if(searchText === '') return this.state.allDatabaseQuestions;
+    const currentQuestions = this.state.allDatabaseQuestions.currentQuestions
+      .filter(question => this.searchQuestionForText(searchText, question))
+      .sort(question => this.countTextOccurrences(searchText, question));
+    const archivedQuestions = this.state.allDatabaseQuestions.archivedQuestions
+      .filter(question => this.searchQuestionForText(searchText, question))
+      .sort(question => this.countTextOccurrences(searchText, question));
+    return {currentQuestions, archivedQuestions};
+  },
+
+  onQuestionsReceived(err, response){
+    const questions = JSON.parse(response.text).questions;
+    if(questions !== undefined){
+      this.setState({loaded: true, allDatabaseQuestions: questions});
+      return;
+    }
+    this.setState({loaded: true});
+  },
+
   onNewQuestion(){
     Routes.navigate(Routes.messagePopupAuthorQuestionsNewPath());
   },
 
+  onQuestionRestore(){
+    const {selectedArchivedQuestion} = this.state;
+    if(selectedArchivedQuestion !== null){
+      const allDatabaseQuestions = _.clone(this.state.allDatabaseQuestions);
+      _.remove(allDatabaseQuestions.archivedQuestions, question => question.id === selectedArchivedQuestion.id);
```

Comment:
I'm not sure this matters in terms of performance or memory usage, and this may just be a style preference, but   I'd suggest using transformations over mutations whenever you can (here that would be `.filter`and `.concat` instead of `_.clone`, `_.remove`, `.push`.